src/spec.c

```diff
@@ -1188,41 +1191,59 @@ static void IndexStats_RdbSave(RedisModuleIO *rdb, IndexStats *stats) {
 
 static threadpool reindexPool = NULL;
 
-typedef struct IndexesScanner {
-  const char *spec_name_opt;  // can be null
-  IndexSpec *spec_opt;
-  size_t scannedKeys, totalKeys;
-} IndexesScanner;
-
-static IndexesScanner *IndexesScanner_New(IndexSpec *spec_opt) {
+static IndexesScanner *IndexesScanner_New(IndexSpec *spec) {
+  if (!spec && global_spec_scanner) {
```

Comment:
Why do you need this? can this even happen?

Reply:
Systematically speaking, you're right: it's not likely we're gonna trigger a global scan while another one is in progress.
But should we have a constructor-level protection? I say why not, and maybe also log about it.